### DISCUSSION TITLE
MET Plus PB2NC Previous File Option

### DISCUSSION THREAD

**Discussion by craigrj** [dtcenter/METplus#1729] [Configuration] [answered]
Is there an ability to add the previous prepbufr file along with the current in the call to PB2NC like you can do in MET?  In MET for some ob times like 3Z, we would need to pull data from the 00Z and the 6Z file.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@craigrj sorry for the delay in getting to this discussion. There certainly already is logic that exists in the METplus PB2NC wrapper to add the `-pbfile` command line argument to the command line. The question is how to trigger that logic.

You can see reference to `-pbfile` in the PB2NC wrapper on this [line of code](https://github.com/dtcenter/METplus/blob/b2b54f5de1005e5a0a29561b43e0ce431c3d534b/metplus/wrappers/pb2nc_wrapper.py#L341). However checking the [PB2NC Wrapper Documentation](https://metplus.readthedocs.io/en/latest/Users_Guide/wrappers.html#pb2nc), I see no mention of that detail.

@georgemccabe I see that the wrapper adds additional input files using the `-pbfile` command line option, but it isn't immediately clear to me how those multiple input files are configured via METplus. Searching through the options, I presume that [PB2NC_FILE_WINDOW_BEGIN](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-PB2NC_FILE_WINDOW_BEGIN) and [PB2NC_FILE_WINDOW_END](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-PB2NC_FILE_WINDOW_END) define the retention time window. So setting:
```
PB2NC_FILE_WINDOW_BEG = -3 * 60 *60 
PB2NC_FILE_WINDOW_END =  3 * 60 * 60
```
Would define a +/- 3 hour time window around the valid time being evaluated. And all the PB2NC input files falling inside that window would automatically be included in the call to PB2NC with the `-pbfile` option? But I don't actually see any examples of those config options being used the METplus repo.

1. @georgemccabe can you confirm whether or not my assumptions are correct?
2. @georgemccabe, if so, I'd recommend adding some info about how these settings relate to the use of the `-pbfile` command line option for pb2nc.

Thanks!

**Comment by georgemccabe** [comment]
@JohnHalleyGotway's assumptions are correct regarding how to pass additional input files to pb2nc. Users can define a window of time to include files using PB2NC_FILE_WINDOW_BEG and PB2NC_FILE_WINDOW_END. The path specified with PB2NC_INPUT_TEMPLATE (and optionally PB2NC_INPUT_DIR) is used to find matching files and extract the timing information based on the filename templates used. Any file that falls within the file window around the valid time will be included in the call to pb2nc. Another way to pass multiple files to pb2nc is to provide a comma-separated list of file path templates in PB2NC_INPUT_TEMPLATE. 

If multiple input files are provided to pb2nc, then any additional file paths besides the first file will be preceded with the `-pbfile` argument. The PB2NC wrapper obfuscated the details of how to build the command regarding the `-pbfile` argument, but I understand that this information would be useful in the METplus User's Guide for users who are moving their existing MET commands into a METplus use case.

I created GitHub issue #1736 to add this information to the user's guide. The Python Wrappers page could be expanded to include specific details like this for each wrapper, so I added a note in this issue to include more details like this for all wrappers.
